portal/src/graphql/portal/AuthenticationLoginIDSettings.tsx

```diff
@@ -1,5 +1,6 @@
-import React, { useContext, useMemo, useState } from "react";
+import React, { useCallback, useContext, useMemo, useState } from "react";
 import produce from "immer";
+import { WritableDraft } from "immer/dist/internal";
```

Comment:
Avoid importing from internal files. I suppose for our purpose, `WritableDraft<T>` should be equivalent to `T`?

Reply:
Using type without `WritableDraft` result in type error `not assignable to type Record<string, unknown>`, `index signature is missing in type ...` in one of the function mutating the fields

Reply:
Object with interface defined is different from general `Object`?

Reply:
The function `setFieldIfListNonEmpty` has a signature too general, we can correct it to this:
```ts
function setFieldIfListNonEmpty<T, K extends keyof T>(
  map: T,
  field: K,
  list: T[K] extends unknown[] | undefined ? T[K] : never,
)
```

Reply:
Thanks, it works